integrations/visual_studio/creating_packages.rst

```diff
@@ -0,0 +1,245 @@
+message.
+
+In the repository, there is already a ``conanfile.py`` recipe:
+
+.. code-block:: python
+
+    from conans import ConanFile, MSBuild
+
+    class HelloConan(ConanFile):
+        name = "hello"
+        version = "0.1"
+        license = "MIT"
+        url = "https://github.com/memsharded/hello_vs"
+        settings = "os", "compiler", "build_type", "arch"
+        exports_sources = "src/*", "build/*"
+
+        def build(self):
+            msbuild = MSBuild(self)
+            msbuild.build("build/HelloLib/HelloLib.sln")
+
+        def package(self):
+            self.copy("*.h", dst="include", src="src")
+            self.copy("*.lib", dst="lib", keep_path=False)
+
+        def package_info(self):
+            self.cpp_info.libs = ["HelloLib"]
+
+
+This recipe is using the (:ref:)`MSBuild() build helper <msbuild>` to build the ``sln`` project.
+If our recipe has ``requires``, the ``MSBUILD`` helper will also take care of inject all the needed
+information from the requirements, as include directories, library names, definitions, flags etc
+to allow our project to locate the declared dependencies.
+
+The recipe contains also a ``test_package`` folder with a simple example consuming application.
+In this example, the consuming application is using CMake to build, but it could also use Visual
+Studio too. We have left the CMake one because it is the default generated with
+:command:`conan new`, and also to show that packages created from Visual Studio projects can also
+be consumed with other build systems like CMake.
+
+Once we want to create a package, it is advised to close VS IDE, clean the temporary build files
+from VS to avoid problems, then create and test the package. Here it is using system defaults,
+assuming they are Visual Studio 14, Release, x86_64:
+
+.. code-block:: bash
+
+   # close VS
+   $ git clean -xdf
+   $ conan create . memsharded/testing
+   ...
+   > Hello World Release!
+
+Instead of closing the IDE and running the command:`git clean` we could also configure a smarter
+filter in ``exports_sources`` field, so temporary build files are not exported into the recipe.
+
+This process can be repeated to create and test packages for different configurations:
+
+.. code-block:: bash
+
+   $ conan create . memsharded/testing -s arch=x86
+   $ conan create . memsharded/testing -s compiler="Visual Studio" -s compiler.runtime=MDd -s build_type=Debug
+   $ conan create . memsharded/testing -s compiler="Visual Studio" -s compiler.runtime=MDd -s build_type=Debug -s arch=x86
+
+.. note::
+
+    It is not mandatory to specify the ``compiler.runtime`` setting. If it is not explicitly
+    defined, Conan will automatically use ``runtime=MDd`` for ``build_type==Debug`` and
+    ``runtime=MD`` for ``build_type==Release``.
+
+
+You can list the different created binary packages:
+
+.. code-block:: bash
+
+    $ conan search hello/0.1@memsharded/testing
+
+Uploading binaries
```

Comment:
This section can be dropped, nothing really useful or specific here

Reply:
Dropped